### PR TITLE
[PWA-713] Update Venia's filters

### PR DESCRIPTION
**Pull request by anthoula**
## Description

Adjust `FILTERS` modal styles to match the new design

## Related Issue
https://jira.corp.magento.com/browse/PWA-713

## Acceptance 
- Verify filters modal matches new design
- Ensure there is no broken functionality with the styling change

### Verification Stakeholders
@dpatil-magento 
@schensley 
@jimbo 

### Verification Steps

1. Go to a category page
2. Click on `Filter` button
3. View the new design for the Filter drawer
4. Verify all inputs are positioned and relevant labels are readable
5. Verify current filter pills display as expected
6. Verify `Clear all` button clears current filter list
7. Verify filters apply as expected in the product results

## Checklist

- [x] I have added tests to cover my changes, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by devops-pwa-codebuild** [comment]
**Performance Test Results**

The following fails have been reported by WebpageTest. These numbers indicates a possible performance issue with the PR which requires further manual testing to validate.

https://pr-2559.pwa-venia.com : LH Performance Expected 0.85 Actual 0.52, LH Best Practices Expected 1 Actual 0.92, WPT Cache Expected 90 Actual 80
https://pr-2559.pwa-venia.com/venia-tops.html : LH Performance Expected 0.75 Actual 0.35, LH Best Practices Expected 1 Actual 0.92
https://pr-2559.pwa-venia.com/valeria-two-layer-tank.html : LH Performance Expected 0.8 Actual 0.4, LH Accessibility Expected 0.9 Actual 0.89, LH Best Practices Expected 1 Actual 0.92, WPT Cache Expected 65 Actual 50

**Comment by schensley** [comment]
@anthoula Things are looking really good with the filters, there are just a couple of things to address.
1. The last section of filters is missing a divider line at the bottom to "contain" the expandable section.
2. The Style Guide recommends the use of a Secondary button for "See Results"
3. Prefer the Size filters be ordered from smallest to largest to be consistent with how they are ordered on the Product Details pages.
4. When the filters are applied and the filter panel closed there isn't any indication that the page has been filtered and which filters have been applied. Even if there were a technical issue preventing the user from removing the filters from this page view, there should be some reference for the shopper without having to open the filter drawer.

**Comment by anthoula** [comment]
> @anthoula Things are looking really good with the filters, there are just a couple of things to address.
> 
> 1. The last section of filters is missing a divider line at the bottom to "contain" the expandable section.

@schensley Added. Noted that this causes a double-line scenario when filters are expanded enough to reach the bottom

> 2. The Style Guide recommends the use of a Secondary button for "See Results"

The button should remain primary, as the preferred action

> 3. Prefer the Size filters be ordered from smallest to largest to be consistent with how they are ordered on the Product Details pages.
> 4. When the filters are applied and the filter panel closed there isn't any indication that the page has been filtered and which filters have been applied. Even if there were a technical issue preventing the user from removing the filters from this page view, there should be some reference for the shopper without having to open the filter drawer.

These are items in the backlog that need some grooming

**Comment by schensley** [comment]
This meets the visual and interaction styles for UX.

File: packages/venia-ui/lib/components/FilterModal/CurrentFilters/currentFilter.js [hunk 1]
```diff
@@ -8,8 +8,7 @@ import Trigger from '../../Trigger';
 import defaultClasses from './currentFilter.css';
 
 const CurrentFilter = props => {
-    const { group, groupName, item, removeItem } = props;
```
Comment (sirugh) [inline packages/venia-ui/lib/components/FilterModal/CurrentFilters/currentFilter.js:11]: You can remove `groupName` from the parent, `CurrentFilters`. And if you do that, I believe you can get rid of `filterNames` being passed in from `FilterModal`.

File: packages/venia-ui/lib/components/FilterModal/filterModal.js [hunk 2]
```diff
@@ -58,12 +59,20 @@ const FilterModal = props => {
         [filterApi, filterItems, filterNames, filterState]
     );
 
+    const clearAll = filterState.size ? (
+        <div className={classes.action}>
+            <LinkButton type="button" onClick={handleReset}>
```
Comment (sirugh) [inline packages/venia-ui/lib/components/FilterModal/filterModal.js:64]: This button needs a focus style/outline.

[Image from Gyazo](https://gyazo.com/75796a523e8bca322323d996c91f5598)

File: packages/venia-ui/lib/components/LinkButton/linkButton.css [hunk 3]
```diff
@@ -6,7 +6,6 @@
     text-decoration: underline;
     line-height: 1.25rem;
     max-width: 100%;
-    outline: none;
```
Comment (sirugh) [inline packages/venia-ui/lib/components/LinkButton/linkButton.css:9]: @davemacaulay was this set to `none` for a reason in the Buttons PR?

Comment (davemacaulay) [reply packages/venia-ui/lib/components/LinkButton/linkButton.css:9]: I think it was incorrectly inherited, so fine to remove here.

**Comment by anthoula** [comment]
> This looks good to me. Did we decide to stick with `uppercase` for the title?

@jimbo Yes, I believe we are going with uppercase, as it matches the XD, and the current Main Menu drawer

**Comment by soumya-ashok** [comment]
@anthoula Sentence case flows better with the new design. Could we do that here? I'll update the styleguide XD.

**Comment by soumya-ashok** [comment]
@anthoula As discussed, the filters work is UX approved, looks great!

A few things to note:

- Need to open a follow-up ticket to address the min-width on the filter and sort buttons (I feel like Dave did have that done, but we can check with him)

- The focus state on the accordion is inconsistent between browser versions, possibly

- The size filter order is being addressed in a follow-up

- Check on possible bug where the applied filter pills don't rearrange when one is removed from the middle.

**Comment by dpatil-magento** [comment]
QA Approved.

**Pull request merged**